### PR TITLE
GH-34956: [Docs][Python] Add to docs the usage of the FixedShapeTensorType

### LINKED ISSUE #34956: [Docs][Python] Add to docs the usage of the FixedShapeTensorType
### Describe the enhancement requested

With the Python bindings for the `FixedShapeTensorType` being worked on in https://github.com/apache/arrow/pull/34883, we should add examples to the PyArrow user guide.

### Component(s)

Python

### PR DESCRIPTION
**Pull request by AlenkaF**
### Rationale for this change

This PR adds examples of the use of `FixedShapeTensorType`to the PyArrow user guide. Should be reviewed and merged after https://github.com/apache/arrow/pull/34883 is done.
* Closes: #34956

### CODE REVIEW

File: docs/source/python/extending_types.rst [hunk 1]
```diff
@@ -357,3 +357,116 @@ pandas ``ExtensionArray``. This method should have the following signature::
...
 This way, you can control the conversion of a pyarrow ``Array`` of your pyarrow
 extension type to a pandas ``ExtensionArray`` that can be stored in a DataFrame.
+
+
+Canonical extension types
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In the :ref:`format_canonical_extensions` under the section **Official List**
+there is a list of canonical extension types. Here you can find an example of how
+to implement and how to use the listed canonical extension types.
```
Comment (rok) [inline docs/source/python/extending_types.rst:367]: Implement seems too strong here? :D
```suggestion
there is a list of canonical extension types. Here we list examples of how to
use the listed canonical extension types.
```

**Review by rok — CHANGES_REQUESTED** [review]
Just two nits.

**Review by rok — APPROVED** [review]
LGTM!

**Comment by AlenkaF** [comment]
I am currently adding some more text. Will probably have extra content ready tomorrow morning!

**Review by jorisvandenbossche — COMMENTED** [review]
Looks good! Some small comments. 

I understand that we initially added this to extending_types.rst (when it was an example of implementing it). But now this is actually part of pyarrow, I am wondering if we would want to move it to a different location in our docs. 
Currently, other data types are mentioned at https://arrow.apache.org/docs/dev/python/data.html, but that's already a long page about a lot of things, so also wouldn't just add it there. Maybe for now we can keep it here, and can think about reorganizing it later.

**Comment by AlenkaF** [comment]
> I understand that we initially added this to extending_types.rst (when it was an example of implementing it). But now this is actually part of pyarrow, I am wondering if we would want to move it to a different location in our docs.
> Currently, other data types are mentioned at https://arrow.apache.org/docs/dev/python/data.html, but that's already a long page about a lot of things, so also wouldn't just add it there. Maybe for now we can keep it here, and can think about reorganizing it later.

I will open an issue for this as I agree it should be reorganised (and I just remembered we have talked about this before, sorry).

Otherwise I have included the changes with the last two commits.

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by AlenkaF** [comment]
Thank you Joris and Rok for reviews!
The issue to rearrange Data Types user guide section can be found here: https://github.com/apache/arrow/issues/35064

**Pull request merged**